src/server/api/routers/traces.ts

```diff
@@ -275,4 +275,37 @@ export const traceRouter = createTRPCRouter({
         observations: enrichedObservations as ObservationReturnType[],
       };
     }),
+  delete: protectedGetTraceProcedure
+    .input(z.object({ traceId: z.string() }))
+    .mutation(async ({ input, ctx }) => {
+      const trace = await ctx.prisma.trace.findFirstOrThrow({
+        where: {
+          id: input.traceId,
+        },
+        include: {
+          scores: true
+        }
+      })
+
+      await trace.scores.map(async (score) => {
```

Comment:
Scores cascade, need to only delete all observations manually

Reply:
ok... no problem